Submitted By:            Igor Živković <contact@igor-zivkovic.from.hr>
Date:                    2015-01-04
Initial Package Version: git20150102
Upstream Status:         Not submitted
Origin:                  Self
Description:             Fixes build with LibreSSL.

--- elinks.orig/configure.in	2015-01-04 23:11:26.010283078 +0100
+++ elinks/configure.in	2015-01-04 23:22:50.212832740 +0100
@@ -1112,9 +1112,11 @@
 			else
 				LIBS="-lssl -lcrypto $LIBS_X"
 			fi
-			AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <openssl/ssl.h>]], [[OpenSSL_add_all_algorithms()]])],[cf_result=yes],[cf_result=no])
+			AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <stdint.h>
+#include <openssl/ssl.h>]], [[OpenSSL_add_all_algorithms()]])],[cf_result=yes],[cf_result=no])
 			if test "$cf_result" != yes; then
-				AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <openssl/ssl.h>]], [[SSLeay_add_ssl_algorithms()]])],[cf_result=yes],[cf_result=no])
+				AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <stdint.h>
+#include <openssl/ssl.h>]], [[SSLeay_add_ssl_algorithms()]])],[cf_result=yes],[cf_result=no])
 			fi
 		fi
 	done
--- elinks.orig/src/network/ssl/socket.c	2015-01-04 23:11:26.123278054 +0100
+++ elinks/src/network/ssl/socket.c	2015-01-04 23:25:37.323386889 +0100
@@ -5,6 +5,7 @@
 #endif
 
 #ifdef CONFIG_OPENSSL
+#include <stdint.h>
 #include <openssl/ssl.h>
 #include <openssl/x509v3.h>
 #define USE_OPENSSL
--- elinks.orig/src/network/ssl/ssl.c	2015-01-04 23:11:26.123278054 +0100
+++ elinks/src/network/ssl/ssl.c	2015-01-04 23:32:28.197066844 +0100
@@ -5,6 +5,7 @@
 #endif
 
 #ifdef CONFIG_OPENSSL
+#include <stdint.h>
 #include <openssl/ssl.h>
 #include <openssl/rand.h>
 #define USE_OPENSSL
@@ -84,11 +85,8 @@
 	 * cannot initialize the PRNG and so every attempt to use SSL fails.
 	 * It's actually an OpenSSL FAQ, and according to them, it's up to the
 	 * application coders to seed the RNG. -- William Yodlowsky */
-	if (RAND_egd(RAND_file_name(f_randfile, sizeof(f_randfile))) < 0) {
-		/* Not an EGD, so read and write to it */
 		if (RAND_load_file(f_randfile, -1))
 			RAND_write_file(f_randfile);
-	}
 
 	SSLeay_add_ssl_algorithms();
 	context = SSL_CTX_new(SSLv23_client_method());
